### PR TITLE
Added spring cloud and dependency injection samples

### PR DESCRIPTION
**Pull request by shreyas-gopalakrishna**
Added spring cloud and dependency injection samples. Updated samples folder structure

Existing samples are moved inside the triggers-bindings folder 
Spring cloud and dependency injection samples are added in their respective folders 

Todo:
- add comments on samples for better readability 
- remove any unnecessary files

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,41 +9,44 @@ description: "This repository contains sample for Azure Functions in Java"
...
+| [ServiceBusQueueTrigger](./triggers-bindings/src/main/java/com/functions/ServiceBusQueueTriggerFunction.java) | ServiceBusQueueTrigger to read message from a queue in service bus and output to service bus queue using ServiceBusQueueOutput binding.  |
+| [ServiceBusTopicTrigger](./triggers-bindings/src/main/java/com/functions/ServiceBusTopicTriggerFunction.java) | ServiceBusTopicTrigger to read message from a topic in service bus and output to service bus topic using ServiceBusTopicOutput binding.  |
+| [Table function](./triggers-bindings/src/main/java/com/functions/TableFunction.java) | Basic example to read and write to table in Azure Storage using TableInput and TableOutput binding.  |
+| [Durable function](./triggers-bindings/src/main/java/com/functions/DurableFunction.java) | Durable function example to start an orchestration and follow activity chaining.  |
+| [Distributed tracing](./triggers-bindings/src/main/java/com/functions/DistributedTracingFunction.java) | Distributed tracing example that adds entries to a MYSQL database via JDBC, adds custom metrics to the Application Insights and outputs to cosmos DB CosmosDBOutput binding.  |
```
[the remark this answers is not in the document]
Comment (shreyas-gopalakrishna) [reply README.md]: Yes, updated durable as well

File: distributed-tracing/host.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+  "version": "2.0",
+  "extensionBundle": {
+    "id": "Microsoft.Azure.Functions.ExtensionBundle",
+    "version": "[3.*, 4.0.0)"
```
Comment (kamperiadis) [inline distributed-tracing/host.json:5]:
```suggestion
    "version": "[4.*, 5.0.0)"
```

File: durable-function/host.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
+  "version": "2.0",
+  "extensionBundle": {
+    "id": "Microsoft.Azure.Functions.ExtensionBundle",
+    "version": "[3.*, 4.0.0)"
```
Comment (kamperiadis) [inline durable-function/host.json:5]:
```suggestion
    "version": "[4.*, 5.0.0)"
```

File: distributed-tracing/pom.xml [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+            <scope>test</scope>
+        </dependency>
+
+        <dependency>
+            <groupId>org.mockito</groupId>
+            <artifactId>mockito-core</artifactId>
+            <version>2.23.4</version>
+            <scope>test</scope>
+        </dependency>
+    </dependencies>
```
Comment (kamperiadis) [inline distributed-tracing/pom.xml:40]:
```suggestion
        <dependency>
            <groupId>com.mysql</groupId>
            <artifactId>mysql-connector-j</artifactId>
            <version>8.1.0</version>
        </dependency> 
       
        <dependency>
            <groupId>com.microsoft.sqlserver</groupId>
            <artifactId>mssql-jdbc</artifactId>
            <version>12.4.1.jre11</version>
        </dependency>

        <dependency>
            <groupId>io.opentelemetry</groupId>
            <artifactId>opentelemetry-api</artifactId>
            <version>1.31.0</version>
        </dependency>

    </dependencies>
```

**Review by kamperiadis — CHANGES_REQUESTED** [review]

**Review by kamperiadis — APPROVED** [review]
LGTM! Thank you

**Pull request merged**